packages/ui/styles/mixins.less

```diff
@@ -22,56 +22,78 @@
 }
 
 // Applies RTL-compatible start and end position to a selector
-.position-start-end (@start, @end) {
+// Simple "when" here just assumes that if the first argument is a list with more than 1 entry,
+// then units are correct. If it's just 1, then we varify the value is a number (measurement).
+.position-start-end (@trbl; @target: "") when (length(@trbl) > 1) or (isnumber(@trbl)) {
+	.position(@trbl);
+
+	.enact-locale-rtl({
+		.position(
+			// Note: swapping left and right values
+			.extract(@trbl; top)[]
+			.extract(@trbl; left)[]
+			.extract(@trbl; bottom)[]
+			.extract(@trbl; right)[]
+		);
+	}; @target);
+}
+
+.position-start-end (@start; @end; @target: "") when (default()) {
 	left: @start;
 	right: @end;
 
-	:global(.enact-locale-right-to-left) & {
+	.enact-locale-rtl({
 		left: @end;
 		right: @start;
-	}
+	}, @target);
 }
 
 // Applies RTL-compatible start and end margins to a selector
-.margin-start-end(@start; @end; @target) when (@target = parent) {
-    margin-left: @start;
-    margin-right: @end;
-
-    :global(.enact-locale-right-to-left)& {
-        margin-left: @end;
-        margin-right: @start;
-    }
-}
-
-.margin-start-end (@start, @end) {
+.margin-start-end (@trbl; @target: "") when (length(@trbl) > 1) or (isnumber(@trbl)) {
+	margin: @trbl;
+
+	.enact-locale-rtl({
+		margin:
+			// Note: swapping left and right values
+			.extract(@trbl; top)[]
+			.extract(@trbl; left)[]
+			.extract(@trbl; bottom)[]
+			.extract(@trbl; right)[]
+		;
+	}; @target);
+}
+.margin-start-end(@start; @end; @target: "") when (default()) {
 	margin-left: @start;
 	margin-right: @end;
 
-	:global(.enact-locale-right-to-left) & {
+	.enact-locale-rtl({
 		margin-left: @end;
 		margin-right: @start;
-	}
+	}, @target);
 }
 
 // Applies RTL-compatible start and end padding to a selector
-.padding-start-end(@start; @end; @target) when (@target = parent) {
-    padding-left: @start;
-    padding-right: @end;
-
-    :global(.enact-locale-right-to-left)& {
-        padding-left: @end;
-        padding-right: @start;
-    }
-}
-
-.padding-start-end (@start, @end) {
+.padding-start-end (@trbl; @target: "") when (length(@trbl) > 1) or (isnumber(@trbl)) {
+	padding: @trbl;
+
+	.enact-locale-rtl({
+		padding:
+			// Note: swapping left and right values
+			.extract(@trbl; top)[]
+			.extract(@trbl; left)[]
+			.extract(@trbl; bottom)[]
+			.extract(@trbl; right)[]
+		;
+	}; @target);
+}
+.padding-start-end(@start; @end; @target: "") when (default()) {
```

Comment:
Do we need an empty line here?

Reply:
I often "group" mixins together that have the same but different argument signatures by omitting the empty line between them. I can add it if you'd prefer.

Reply:
Nope, I agree with your opinion. Thanks for clarification.